docs/user_guide/productionizing/configuring_access_to_gpus.md

```diff
@@ -6,31 +6,405 @@
 .. tags:: Deployment, Infrastructure, GPU, Intermediate
 ```
 
-Along with the simpler resources like CPU/Memory, you may want to configure and access GPU resources. Flyte
-allows you to configure the GPU access poilcy for your cluster. GPUs are expensive and it would not be ideal to
-treat machines with GPUs and machines with CPUs equally. You may want to reserve machines with GPUs for tasks
-that explicitly request GPUs. To achieve this, Flyte uses the Kubernetes concept of [taints and tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/).
+Along with compute resources like CPU and memory, you may want to configure and access GPU resources. 
 
-Kubernetes can automatically apply tolerations for extended resources like GPUs using the [ExtendedResourceToleration plugin](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#extendedresourcetoleration), enabled by default in some cloud environments. Make sure the GPU nodes are tainted with a key matching the resource name, i.e., `key: nvidia.com/gpu`.
+Flyte provides different ways to request accelerator resources directly from the task decorator. This page covers the requirements and procedures to leverage them. 
 
-You can also configure Flyte backend to apply specific tolerations. This configuration is controlled under generic  k8s plugin configuration as can be found [here](https://github.com/flyteorg/flyteplugins/blob/5a00b19d88b93f9636410a41f81a73356a711482/go/tasks/pluginmachinery/flytek8s/config/config.go#L120).
+>The examples in this section use [ImageSpec](https://docs.flyte.org/en/latest/user_guide/customizing_dependencies/imagespec.html#imagespec), a Flyte feature that builds a custom container image without a Dockerfile. Install it using `pip install flytekitplugins-envd`.
 
-The idea of this configuration is that whenever a task that can execute on Kubernetes requests for GPUs, it automatically
-adds the matching toleration for that resource (in this case, `gpu`) to the generated PodSpec.
-As it follows here, you can configure it to access specific resources using the tolerations for all resources supported by
-Kubernetes.
+## Requesting a GPU with no preference for device
+The goal in this example is to run the task on a single available GPU :
 
-Here's an example configuration:
+```python
+from flytekit import ImageSpec, Resources, task
 
+image = ImageSpec(
+    base_image= "ghcr.io/flyteorg/flytekit:py3.10-1.10.2",
+     name="pytorch",
+     python_version="3.10",
+     packages=["torch"],
+     builder="envd",
+     registry="<YOUR_CONTAINER_REGISTRY>",
+ )
```

Comment:
I do not think this this works with the `envd` image builder. This should work tho:

```python
image = ImageSpec(
    name="pytorch",
    python_version="3.10",
    packages=["torch"],
    builder="default",
    registry=...
)
```